lms/djangoapps/course_home_api/progress/v1/serializers.py

```diff
@@ -51,11 +51,46 @@ class ChapterSerializer(serializers.Serializer):
     subsections = SubsectionSerializer(source='sections', many=True)
 
 
+class CertificateDataSerializer(serializers.Serializer):
+    cert_web_view_url = serializers.CharField()
+    download_url = serializers.CharField()
+    is_downloadable = serializers.SerializerMethodField()
+    is_requestable = serializers.SerializerMethodField()
+    msg = serializers.CharField()
+    request_cert_url = serializers.SerializerMethodField()
+    title = serializers.CharField()
+
+    def get_is_downloadable(self, cert_data):
+        return cert_data.cert_status == CertificateStatuses.downloadable and cert_data.download_url is not None
+
+    def get_is_requestable(self, cert_data):
+        return cert_data.cert_status == CertificateStatuses.requesting and cert_data.request_cert_url is not None
+
+    def get_request_cert_url(self, cert_data):
+        return reverse('generate_user_cert', args=[self.context['course_key']])
```

Comment:
This seems odd. In the check for `is_requestable`, you're seeing if the request_cert_url exists, but then you create it below. So maybe the above check doesn't need it. Or if I'm reading this right, you might not need this get_request_cert_url since the cert_data already has that field defined.